server/mailet/mailets/src/main/java/org/apache/james/transport/mailets/RecipientRewriteTableProcessor.java

```diff
@@ -0,0 +1,313 @@
+            .toList();
+    }
+
+    private ImmutableList<MailAddress> getLocalAddresses(ImmutableList<MailAddress> mailAddresses,
+            final MailetContext context) {
+        return FluentIterable.from(mailAddresses)
+            .filter(isLocalServer(context))
+            .toList();
+    }
+
+    private ImmutableList<MailAddress> buildMailAddressFromMappingAddress(ImmutableList<Mapping> newMappings) {
+        return FluentIterable.from(newMappings)
+            .transform(mailAddressFromMapping)
+            .filter(mailAddressPresent)
+            .transform(mailAddressFromOptional)
+            .toList();
+    }
+
+    private ImmutableList<Mapping> getAddressWithNoDomain(Mappings mappings, DomainList domainList) throws MessagingException {
+        ImmutableList<Mapping> addressWithoutDomains = FluentIterable.from(mappings)
+            .filter(noneDomain)
+            .toList();
+        
+        if (!addressWithoutDomains.isEmpty()) {
+            final String defaultDomain = getDefaultDomain(domainList);
+
+            return FluentIterable.from(addressWithoutDomains)
+                .transform(appendDefaultDomain(defaultDomain))
+                .toList();
+        }
+        return ImmutableList.of();
+    }
+
+    private Function<Mapping, Mapping> appendDefaultDomain(final String defaultDomain) {
+        return new Function<Mapping, Mapping>() {
+            @Override
+            public Mapping apply(Mapping address) {
+                return address.appendDomain(defaultDomain);
+            }
+        };
+    }
+
+    private Predicate<MailAddress> isLocalServer(final MailetContext context) {
+        return new Predicate<MailAddress>() {
+            @Override
+            public boolean apply(MailAddress mailAddress) {
+                return context.isLocalServer(mailAddress.getDomain());
+            }
+        };
+    }
+
+    private Predicate<MailAddress> isNotLocalServer(final MailetContext context) {
+        return new Predicate<MailAddress>() {
+            @Override
+            public boolean apply(MailAddress mailAddress) {
+                return !context.isLocalServer(mailAddress.getDomain());
+            }
+        };
+    }
+
+    private void forwardToRemoteAddress(MailAddress sender, MailAddress recipient, MimeMessage message, ImmutableList<MailAddress> mailAddresses, final MailetContext context) throws MessagingException {
+        ImmutableList<MailAddress> remoteAddress = FluentIterable.from(mailAddresses)
+            .filter(isNotLocalServer(context))
+            .toList();
+
+        if (!remoteAddress.isEmpty()) {
+            try {
+                context.sendMail(sender, remoteAddress, message);
+                context.log(LogLevel.INFO, "Mail for " + recipient + " forwarded to " + remoteAddress);
+            } catch (MessagingException ex) {
+                context.log(LogLevel.INFO, "Error forwarding mail to " + remoteAddress + "attempting local delivery");
```

Comment:
Your message should be : 

```
context.log(LogLevel.INFO, "Error forwarding mail to " + remoteAddress);
```

We should not try to guess from here what will be done. Maybe the admin did not configure LocalDelivery after this.